### DISCUSSION TITLE
the key of this in Schema.path occur error with typecsrcipt

### DISCUSSION THREAD

**Discussion by huge689** [Automattic/mongoose#12557] [Q&A]
code:
```
UserSchema.path('hashed_password').validate(function (v) {
  if (this._password && this._password.length < 8) {
    this.invalidate('password', 'Password must be at least 8 characters.');
  }
});
```
error:
any
'this' implicitly has type 'any' because it does not have a type annotation.ts(2683)
user.model.ts(57, 45): An outer value of 'this' is shadowed by this container.
'this' implicitly has type 'any' because it does not have a type annotation.ts(2683)
user.model.ts(57, 45): An outer value of 'this' is shadowed by this container.
'this' implicitly has type 'any' because it does not have a type annotation.ts(2683)

**Comment by vkarpov15** [comment]
This is a bug in our TypeScript bindings, fix will be in 6.7.2. See https://github.com/Automattic/mongoose/pull/12663
